common/flatpak-dir.c

```diff
@@ -1153,6 +1154,57 @@ flatpak_dir_remove_oci_files (FlatpakDir   *self,
   return TRUE;
 }
 
+static gchar *
+flatpak_dir_revokefs_fuse_create_mountpoint (const gchar *ref,
+                                             GError     **error)
+{
+  g_autoptr(GFile) cache_dir = NULL;
+  g_auto(GStrv) parts = NULL;
+  g_autofree gchar *cache_dir_path = NULL;
+  g_autofree gchar *mnt_dir = NULL;
+  g_autofree gchar *mountpoint = NULL;
+
+  cache_dir = flatpak_ensure_system_user_cache_dir_location (error);
+  if (cache_dir == NULL)
+    return NULL;
+
+  parts = flatpak_decompose_ref (ref, NULL);
+  cache_dir_path = g_file_get_path (cache_dir);
+  mnt_dir = g_strdup_printf ("%s-XXXXXX", parts[1]);
+  mountpoint = g_mkdtemp_full (g_build_filename (cache_dir_path, mnt_dir, NULL), 0755);
+
+  return g_steal_pointer (&mountpoint);
+}
+
+static gboolean
+flatpak_dir_revokefs_fuse_unmount (OstreeRepo **repo,
+                                  GLnxLockFile *lockfile,
+                                  const gchar *mnt_dir,
+                                  GError **error)
+{
+  g_autoptr(GSubprocess) fusermount = NULL;
+
+  /* Clear references to child_repo as not to leave any open fds. This is needed for
+   * a clean umount operation.
+   */
+  g_clear_pointer (repo, g_object_unref);
+  glnx_release_lock_file (lockfile);
+
+  fusermount = g_subprocess_new (G_SUBPROCESS_FLAGS_NONE,
+                                 error,
+                                 "fusermount", "-u", mnt_dir,
+                                 NULL);
```

Comment:
I think you should pass -z here to. That makes it more likely that the unmount succeeds, and I don't think there are any security aspects of it, given that we're guaranteed to close the writer socket.

Reply:
In fact, it may fix the issue @wjt mentions with the lock not being held.

Reply:
Introduced `-z` flag in the subprocess